src/Hangfire.PostgreSql/PostgreSqlFetchedJob.cs

```diff
@@ -20,55 +20,85 @@
 //    Special thanks goes to him.
 
 using System;
+using System.Threading;
 using Dapper;
+using Hangfire.Logging;
+using Hangfire.PostgreSql.Utils;
 using Hangfire.Storage;
 
 namespace Hangfire.PostgreSql
 {
   public class PostgreSqlFetchedJob : IFetchedJob
   {
+    private readonly ILog _logger = LogProvider.GetLogger(typeof(PostgreSqlFetchedJob));
+    
     private readonly PostgreSqlStorage _storage;
     private bool _disposed;
     private bool _removedFromQueue;
     private bool _requeued;
 
+    private readonly object _syncRoot = new object();
+    private long _lastHeartbeat;
+    private TimeSpan _interval;
+    
     public PostgreSqlFetchedJob(
       PostgreSqlStorage storage,
       long id,
       string jobId,
-      string queue)
+      string queue,
+      DateTime? fetchedAt)
     {
       _storage = storage ?? throw new ArgumentNullException(nameof(storage));
 
       Id = id;
       JobId = jobId ?? throw new ArgumentNullException(nameof(jobId));
       Queue = queue ?? throw new ArgumentNullException(nameof(queue));
+      FetchedAt = fetchedAt ?? throw new ArgumentNullException(nameof(fetchedAt));
+
+      if (storage.Options.SlidingInvisibilityTimeout.HasValue)
+      {
+        _lastHeartbeat = TimestampHelper.GetTimestamp();
+        _interval = TimeSpan.FromSeconds(storage.Options.SlidingInvisibilityTimeout.Value.TotalSeconds / 5);
+        storage.HeartbeatProcess.Track(this);
+      }
     }
 
     public long Id { get; }
     public string Queue { get; }
     public string JobId { get; }
+    internal DateTime? FetchedAt { get; private set; }
 
     public void RemoveFromQueue()
     {
-      _storage.UseConnection(null, connection => connection.Execute($@"
-        DELETE FROM ""{_storage.Options.SchemaName}"".""jobqueue"" WHERE ""id"" = @Id;
+      lock (_syncRoot)
+      {
+        if (!FetchedAt.HasValue) return;
+        
+        _storage.UseConnection(null, connection => connection.Execute($@"
+        DELETE FROM ""{_storage.Options.SchemaName}"".""jobqueue"" WHERE ""id"" = @Id AND ""fetchedat"" = @FetchedAt;
```

Comment:
Feels like formatting is off. Pretty sure `.editorconfig` has rules to always add brackets around single lines (e.g. `return;` on line 75, and the string here is not indented right.

Reply:
I use Rider and perhaps it isn't following the .editorconfig properly.  The ```if (!FetchedAt.HasValue) return;``` I copied from the SQL Server storage project directly.  

I'll fix this up and check to see if other changes need tidying up as well.